### PR TITLE
feat(server): set up config

### PR DESCRIPTION
**Pull request by VirgileLyon**
You can set up your config 🖌  : 
- Regex
- Name of gauge
- Labels and tags

Capture d’écran 2019-11-29 à 17 34 53

### CODE REVIEW

File: config/default.json.dist [hunk 1]
```diff
@@ -6,5 +6,10 @@
       "baseUrl": "your URL",
       "route": "'your route'",
       "listener": "your listener"
+      "filterby": "your filter",
+      "nameofloadtimegauge": "name of your loadtime gauge",
```
Comment (Slashgear) [inline config/default.json.dist]: I would prefer `loadTimeGaugeName` in 🐫  case 
Same for the other ones

File: src/index.js [hunk 2]
```diff
@@ -4,24 +4,29 @@ const fetchStatuCakeData = require('./statuscake');
...
   help: 'Loadtime of your page',
-  labelNames: ['customer', 'service', 'device', 'version'],
+  labelNames: (tags),
 });
 const filesizeGauge = new client.Gauge({
-  name: 'frontend_pagespeed_filesize_bytes',
+  name: (config.statuscake.nameoffilesizegauge),
   help: 'Filesize of your page',
-  labelNames: ['customer', 'service', 'device', 'version'],
+  labelNames: (tags),
```
Comment (Slashgear) [inline src/index.js]: Putting `()` arround variables in JS doesn't mean something. 
We should avoid that 🚥

File: src/index.js [hunk 3]
```diff
@@ -4,24 +4,29 @@ const fetchStatuCakeData = require('./statuscake');
 const server = require('./server');
 const storage = require('./statuscakedata');
 
-const regex = /\s?([-])\s?/;
+const regexp = config.statuscake.regex;
```
Comment (Slashgear) [inline src/index.js]:
```suggestion
const  { regexp } = config.statuscake;
```

File: src/index.js [hunk 4]
```diff
@@ -4,24 +4,29 @@ const fetchStatuCakeData = require('./statuscake');
...
 
-const getTagsFromTitle = Title => Title.split(regex);
+const getTagsFromTitle = title => title.match(new RegExp(regexp));
```
Comment (Slashgear) [inline src/index.js]: Are we sure this function is doing want it's name means ?

File: config/default.json.dist [hunk 5]
```diff
@@ -6,5 +6,10 @@
...
+      "nameoffilesizegauge": "name of your filesize gauge",
+      "nameofrequestgauge": "name of your request gauge",
+      "regex": "Your regex"
```
Comment (fdubost) [inline config/default.json.dist]: `regexp` is too much generic, prefer something like `tagsFromTestTitleRegexp` 👯

File: src/index.js [hunk 6]
```diff
@@ -4,24 +4,29 @@ const fetchStatuCakeData = require('./statuscake');
...
+const tags = [
+  (config.statuscake.customer),
```
Comment (fdubost) [inline src/index.js]: 🙅‍♂ Nop, this tags array is not generic. Tag list should be declared in the config 😇

File: config/default.json.dist [hunk 7]
```diff
@@ -6,5 +6,10 @@
       "baseUrl": "your URL",
       "route": "'your route'",
       "listener": "your listener"
+      "filterby": "your filter",
```
Comment (fdubost) [inline config/default.json.dist]: `testTitleFilter`? 🤷‍♂ (a page speed item in statuscake is a test https://www.statuscake.com/api/Page%20Speed/List%20Pagespeed%20Test.md)

File: config/default.json.dist (hunk 1 above)
Comment (VirgileLyon) [reply config/default.json.dist]: 👌

File: src/index.js (hunk 2 above)
Comment (VirgileLyon) [reply src/index.js]: 🤜

File: config/default.json.dist (hunk 5 above)
Comment (VirgileLyon) [reply config/default.json.dist]: ✌️

File: config/default.json.dist (hunk 7 above)
Comment (VirgileLyon) [reply config/default.json.dist]: ☝️

File: src/index.js (hunk 6 above)
Comment (VirgileLyon) [reply src/index.js]: Exact 🐙

File: src/index.js (hunk 4 above)
Comment (VirgileLyon) [reply src/index.js]: Exact 🍂

File: src/index.js (hunk 3 above)
Comment (VirgileLyon) [reply src/index.js]: 👍

File: config/default.json.dist [hunk 8]
```diff
@@ -6,5 +6,11 @@
       "baseUrl": "your URL",
       "route": "'your route'",
       "listener": "your listener"
+      "testTitleFilter": "your filter",
+      "nameOflLoadTimeGauge": "name of your loadtime gauge",
+      "nameOfFileSizeGauge": "name of your filesize gauge",
+      "nameOfRequestGauge": "name of your request gauge",
+      "tagsFromTestTitleRegexp": "Your regex"
+      "tags": ["Your first tag", "Your second tag"]
```
Comment (fdubost) [inline config/default.json.dist]: Don't forget to change the docs 😉

Comment (VirgileLyon) [reply config/default.json.dist]: For the next PR 😸

File: src/index.js [hunk 9]
```diff
@@ -4,36 +4,33 @@ const fetchStatuCakeData = require('./statuscake');
 const server = require('./server');
 const storage = require('./statuscakedata');
 
-const regex = /\s?([-])\s?/;
+const { tagsFromTestTitleRegexp } = config.statuscake;
```
Comment (florentbarriol) [inline src/index.js]:
```suggestion
const { tagsFromTestTitleRegexp, nameOfLoadTimeGauge, tags, nameOfFileSizeGauge, nameOfRequestGauge } = config.statuscake;
```
You can get all the varaibles from statuscake here

File: config/default.json.dist [hunk 10]
```diff
@@ -6,5 +6,11 @@
       "baseUrl": "your URL",
       "route": "'your route'",
       "listener": "your listener"
+      "testTitleFilter": "your filter",
+      "nameOflLoadTimeGauge": "name of your loadtime gauge",
```
Comment (piouPiouM) [inline config/default.json.dist]: Extra `l` 👁 
```suggestion
      "nameOfLoadTimeGauge": "name of your load time gauge",
```

Comment (VirgileLyon) [reply config/default.json.dist]: 👀

File: src/index.js (hunk 9 above)
Comment (VirgileLyon) [reply src/index.js]: Thanks ✌️

**Pull request merged**